hail/src/main/scala/is/hail/expr/ir/lowering/LowerTableIR.scala

```diff
@@ -466,6 +467,65 @@ object LowerTableIR {
         invoke("sum", TInt64,
           stage.mapCollect(relationalLetsAbove)(rows => foldIR(mapIR(rows)(row => Consume(row)), 0L)(_ + _)))
 
+      case TableToValueApply(child, TableCalculateNewPartitions(nPartitions)) =>
+        val stage = lower(child)
+        val sampleSize = math.min(nPartitions * 20, 1000000)
+        val samplesPerPartition = sampleSize / stage.numPartitions
```

Comment:
Have to guard against division by zero

Reply:
I stared at this for a bit and I'm not sure what we should do if there are no rows. You'll get failures when you try to pass round an empty array of intervals. For example, ir.TableNativeReader will complain.

Reply:
I feel like we should figure out how to pervasively support empty intervals, but for now maybe we just check and return an empty table in repartition?

Reply:
It looks like we're getting another divide-by-zero, which I think comes from the StreamRange with stepSize=0, which happens when there are no rows. It should be easy to fix to create an empty partitioner in that case. Is there any reason TableNativeReader can't support reading with zero partitions?

Reply:
I'm unaware of a true reason. I might have fixed it in my [most recent commit](https://github.com/hail-is/hail/pull/11665/commits/d2eb207acd4b5053dae62ee8235bfa68ea6aaab4). We'll see what CI says.